source/connection-string/tests/valid-options.yml

```diff
@@ -28,3 +28,17 @@ tests:
         auth: ~
         options:
               tls: true
+    -
+        description: Colon in a key value pair
+        uri: mongodb://example.com?authMechanismProperties=TOKEN_RESOURCE:mongodb%3A%2F%2Ftest-cluster
```

Comment:
```suggestion
        uri: mongodb://example.com?authMechanismProperties=TOKEN_RESOURCE:mongodb://test-cluster
```

Reply:
Ah, this was failing in the python driver, but I see it is because we have a custom check there for `:`, let me try without.